packages/autocomplete-insights-loader/src/index.ts

```diff
@@ -0,0 +1,38 @@
+// @ts-ignore
+import { createAlgoliaInsightsPlugin } from '@algolia/autocomplete-plugin-algolia-insights';
+
+export function loadInsightsPlugin(props: any) {
+  // We assume the logic for downloading automatically Algolia Insights
+  // is defined here and the following code depends on it.
+  const insightsClient = (props.environment.window as Window & {
+    aa: (...args: any[]) => void;
+  }).aa;
+
+  // Iterate over plugins and add algoliaInsights plugin if
+  // - it isn't already added
+  // - existing plugins have a reference to `searchClient` OR
+  // ?: Is there a way to also detect when `getAlgoliaResult()`
+  //    is used in a source?
+  const hasAlgoliaInsightsPlugin = props.plugins.some(
+    (plugin) => plugin.name === 'aa.algoliaInsightsPlugin'
+  );
+  const { __autocomplete_pluginOptions } =
+    props.plugins.find(
+      (plugin) => plugin.__autocomplete_pluginOptions?.searchClient
+    ) || {};
+  const searchClient = __autocomplete_pluginOptions?.searchClient;
+
+  if (!hasAlgoliaInsightsPlugin && searchClient) {
+    insightsClient('init', {
```

Comment:
shouldn't the `init` be part of the plugin itself, like in middleware for InstantSearch?

Reply:
I might backtrack from this, the plugin doesn't know the app id and the api key. We can discuss in RFC if we can justify a change in the lib to make it aware of those credentials.